### ISSUE TITLE
Pub_sub example does not works

### ISSUE THREAD

**Issue by dbelovol** [aws/aws-iot-device-sdk-js-v2#136]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-iot-device-sdk-js-v2/issues) and didn't find any solution
  
**Known Issue**
- [x] I'm using ATS data type endpoint: the endpoint should look like `<prefix>-ats.iot.<region>.amazonaws.com`

**Describe the bug**
I've tried sample code from [here](https://github.com/aws/aws-iot-device-sdk-js-v2/tree/main/samples/node/pub_sub) and it does not works for me.

**SDK version number**
Latest 

**Platform/OS/Hardware/Device**
openSUSE Leap 15.1, 4x Intel Core i7 

**To Reproduce (observed behavior)**
Steps to reproduce the behavior (please share code)
1. git clone https://github.com/aws/aws-iot-device-sdk-js-v2.git
2. cd aws-iot-device-sdk-js-v2/samples/node/pub_sub
3. npm install
4. node dist/index.js --endpoint xxxxxxxxx-ats.iot.eu-central-1.amazonaws.com --ca_file root-CA.crt --cert MyLinuxMachine.cert.pem --key MyLinuxMachine.private.key

**Expected behavior**
As specified in README

**Logs/output**
events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: Failed to connect: aws-c-io: AWS_IO_TLS_ERROR_NEGOTIATION_FAILURE, TLS (SSL) negotiation failed
    at process.nextTick (/home/belovol/DEVEL/AWS_IoT_V2/aws-iot-device-sdk-js-v2/samples/node/pub_sub/node_modules/aws-crt/dist/native/mqtt.js:333:36)
    at process._tickCallback (internal/process/next_tick.js:61:11)
Emitted 'error' event at:
    at MqttClientConnection.emit (/home/belovol/DEVEL/AWS_IoT_V2/aws-iot-device-sdk-js-v2/samples/node/pub_sub/node_modules/aws-crt/dist/common/event.js:75:22)
    at process.nextTick (/home/belovol/DEVEL/AWS_IoT_V2/aws-iot-device-sdk-js-v2/samples/node/pub_sub/node_modules/aws-crt/dist/native/mqtt.js:333:22)
    at process._tickCallback (internal/process/next_tick.js:61:11)

I've tried v1 with exactly the same credentials. Everything worked as expected

**Comment by dbelovol** [comment]
Run under --verbose TRACE and see this
negotiation failed with error Certificate is untrusted (Error encountered in /codebuild/output/src262905832/src/aws-crt-nodejs/crt/s2n/tls/s2n_x509_validator.c line 372)

**Comment by jmklix** [comment]
I believe you are running into a policy issue which can be solved by using a full access policy temporarily. It should lock something like this: 
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "iot:*"
            ],
            "Resource": "*"
        }
    ]
}
```
After you get it working make sure to limit the policy to only the actions and resources that you need. Please let me know if this doesn't work or if you have any more questions about this.

**Comment by dbelovol** [comment]
This does not work. As I've said earlier, I've tried aws-iot-device-sdk-js with exactly the same credentials and everything worked perfecntly. I didn't change neither the  credential nor the policy. 
But I've tried proposed policy - same result. aws-iot-device-sdk-js works, aws-iot-device-sdk-js-v2 - not.
There is next code in line 372 of s2n_x509_validator.c line:
`370 op_code = X509_verify_cert(validator->store_ctx);
371
372         S2N_ERROR_IF(op_code <= 0, S2N_ERR_CERT_UNTRUSTED);`
As stated [here](https://linux.die.net/man/3/x509_verify_cert) :
".. Applications rarely call this function directly but it is used by OpenSSL internally for certificate validation, in both the S/MIME and SSL/TLS code.

The negative return value from X509_verify_cert() can **only** occur if no certificate is set in ctx (due to a programming error) or if a retry operation is requested during internal lookups (which never happens with standard lookup methods). It is however recommended that application check for <= 0 return value on error..."
I've cheked, that certificate has been passed by 
`concole.log(argv)`
